### DISCUSSION TITLE
Equations

### DISCUSSION THREAD

**Discussion by wolkenmachine** [egraphs-good/egg#242] [Q&A] [answered]
Is is possible to define a language that includes an equality symbol?
For example, could I add expressions like this: `(= a (+ b c))` ,
or have rewrite rules like this: `(= ?a (+ ?b ?c))" => "(= ?b (- ?a ?c))`

It seems like it should be possible in principle, perhaps by merging e-classes manually? But, I haven't been able to figure out from the docs how to go about doing this. Thanks for your help!

**Comment by mwillsey** [comment, marked as answer]
Do you mean that the equality symbol should be interpreted as the e-graphs notion of equality? Check out the multi-patterns feature [docs](https://docs.rs/egg/latest/egg/struct.MultiPattern.html) [example](https://github.com/egraphs-good/egg/blob/main/tests/datalog.rs). This will be much better supported in an upcoming version of egg.... stay tuned!!

**Reply by wolkenmachine** [reply]
Great, multi-patterns is exactly what I was looking for. Thank you!
